lib/wallet/account/hot-wallet-account.ts

```diff
@@ -137,29 +142,50 @@ export class HotWalletAccount implements IWallet {
       throw new Error("No transaction to sign");
     }
 
-    // Replace the input script with the provided script
-    if (options?.scriptHex) {
-      const replaceIndex = pending.transaction.inputs.findIndex(
-        (input) => input.signatureScript === "",
-      );
-      if (replaceIndex === -1) {
-        throw new Error("No input to replace");
+    const signed = await this.signTx(pending.transaction, options?.scripts);
+    return (await this.rpcClient.submitTransaction({ transaction: signed }))
+      .transactionId;
+  }
+
```

Comment:
Same comments as for the private key implementation.

Reply:
https://github.com/forbole/kastle/pull/22/commits/49663fe0266ded9027b9b358435c3e9318e406f6